In [1]:
import sys
from pathlib import Path

# Agrega src/ al path para poder importar pipeline.py
sys.path.insert(0, str(Path.cwd().parent / "src"))

from pipeline import construir_serie
from config import PATHS

print("Imports OK")
print(f"Raw_Usados: {PATHS['raw_usados']}")
print(f"Existe: {PATHS['raw_usados'].exists()}")

Imports OK
Raw_Usados: c:\Users\usuario\Documents\PORTFOLIO\mercado-limones-causal\data\Raw_Usados
Existe: True


In [3]:
serie_usados = construir_serie("usados")

CONSTRUYENDO SERIE — USADOS
Archivos encontrados: 41
2023-01 →    133,102 originales →  132,129 limpios (99.3%)
2023-02 →    111,686 originales →  110,737 limpios (99.2%)
2023-03 →    146,851 originales →  145,710 limpios (99.2%)
2023-04 →    130,209 originales →  129,094 limpios (99.1%)
2023-05 →    142,233 originales →  141,023 limpios (99.1%)
2023-06 →    134,148 originales →  133,028 limpios (99.2%)
2023-07 →    154,444 originales →  153,239 limpios (99.2%)
2023-08 →    162,509 originales →  161,155 limpios (99.2%)
2023-09 →    143,838 originales →  142,634 limpios (99.2%)
2023-10 →    141,592 originales →  140,402 limpios (99.2%)
2023-11 →    132,344 originales →  131,209 limpios (99.1%)
2023-12 →    122,055 originales →  120,987 limpios (99.1%)
2024-01 →    116,151 originales →  115,204 limpios (99.2%)
2024-02 →    106,239 originales →  105,408 limpios (99.2%)
2024-03 →    106,001 originales →  105,120 limpios (99.2%)
2024-04 →    136,265 originales →  135,209 limpios (99.2%)
202

In [4]:
print(serie_usados.to_string())

    periodo  anio  mes   total  antiguedad_media  pct_protocolo21  pct_jovenes
0   2023-01  2023    1  132129           14.2097          40.6640       6.8683
1   2023-02  2023    2  110737           14.3191          40.6946       6.9390
2   2023-03  2023    3  145710           14.1501          41.0610       7.3770
3   2023-04  2023    4  129094           14.1324          40.8857       7.3605
4   2023-05  2023    5  141023           14.1860          40.5090       7.2676
5   2023-06  2023    6  133028           14.2124          40.8313       7.3616
6   2023-07  2023    7  153239           14.0721          40.7533       8.0084
7   2023-08  2023    8  161155           14.0803          40.4623       8.2566
8   2023-09  2023    9  142634           14.4119          39.6743       8.1215
9   2023-10  2023   10  140402           14.4660          39.3378       8.2250
10  2023-11  2023   11  131209           14.7482          38.4158       8.5840
11  2023-12  2023   12  120987           14.4577    

In [5]:
output_path = PATHS["processed"] / "serie_usados.parquet"
serie_usados.to_parquet(output_path, index=False)
print(f"Guardado en: {output_path}")

Guardado en: c:\Users\usuario\Documents\PORTFOLIO\mercado-limones-causal\data\processed\serie_usados.parquet


In [6]:
print(f"Shape: {serie_usados.shape}")
print(f"\nPrímer período: {serie_usados['periodo'].iloc[0]}")
print(f"Último período: {serie_usados['periodo'].iloc[-1]}")
print(f"\nSerie completa:")
print(serie_usados.to_string())

Shape: (41, 7)

Prímer período: 2023-01
Último período: 2026-05

Serie completa:
    periodo  anio  mes   total  antiguedad_media  pct_protocolo21  pct_jovenes
0   2023-01  2023    1  132129           14.2097          40.6640       6.8683
1   2023-02  2023    2  110737           14.3191          40.6946       6.9390
2   2023-03  2023    3  145710           14.1501          41.0610       7.3770
3   2023-04  2023    4  129094           14.1324          40.8857       7.3605
4   2023-05  2023    5  141023           14.1860          40.5090       7.2676
5   2023-06  2023    6  133028           14.2124          40.8313       7.3616
6   2023-07  2023    7  153239           14.0721          40.7533       8.0084
7   2023-08  2023    8  161155           14.0803          40.4623       8.2566
8   2023-09  2023    9  142634           14.4119          39.6743       8.1215
9   2023-10  2023   10  140402           14.4660          39.3378       8.2250
10  2023-11  2023   11  131209           14.7482  

## Verificación — Consistencia con proyecto anterior

Los 7 puntos del análisis exploratorio previo (mercado-limones-argentino)
están embebidos en esta serie continua. Cruce exacto:

| Período | Registros anterior | Registros nuevo | Antigüedad anterior | Antigüedad nueva |
|---|---|---|---|---|
| 2023-01 | 132.129 | 132.129 | 14.21 | 14.21 ✅ |
| 2023-10 | 140.402 | 140.402 | 14.47 | 14.47 ✅ |
| 2024-02 | 105.408 | 105.408 | 15.05 | 15.05 ✅ |
| 2024-12 | 159.667 | 159.667 | 13.34 | 13.34 ✅ |
| 2025-01 | 169.213 | 169.213 | 14.39 | 14.39 ✅ |
| 2025-10 | 165.683 | 165.683 | 13.95 | 13.95 ✅ |
| 2026-03 | 153.351 | 153.374 | 14.81 | 14.81 ✅ |

Diferencia en 2026-03: 23 registros (0.015%) — filtro TRAMITES_EXCLUIDOS_USADOS
captura registros que el pipeline anterior no excluía. Negigible y metodológicamente correcto.

## Hallazgos — Lo que la serie continua revela

### 1. Estacionalidad de diciembre — patrón estructural confirmado

Cada diciembre es el mínimo de antigüedad de su año:
- 2023-12: 14.46 → 2024-01: 15.13 (rebote)
- 2024-12: 13.34 → 2025-01: 14.39 (rebote)
- 2025-12: 13.64 → 2026-01: 14.47 (rebote)

Consistente con Jensen & Meckling (1976): presión de objetivos comerciales
de fin de año fuerza liquidación de mejores unidades. No es evento aislado
— es regularidad estructural del mercado. Requiere variable dummy estacional
en el modelo ADL.

### 2. Epicentro del shock Milei — enero 2024, no febrero

2024-01: 15.13 años es el punto más alto de toda la serie.
El diseño muestral del proyecto anterior había elegido 2024-02 como proxy
del shock, pero la serie continua muestra que el pico fue un mes antes.
**Implicancia para el modelo causal:** el lag real del shock puede ser
distinto al estimado con 7 puntos.

### 3. Protocolo 21 — tendencia ascendente secular

Participación promedio por año:
- 2023: ~40.2%
- 2024: ~39.3% (baja transitoria durante shock)
- 2025: ~41.3%
- 2026 (ene-may): ~42.1%

La desregulación de importaciones post-2024 tiene huella clara y sostenida.
Esto refuerza la necesidad de variable de control macroeconómico en el modelo
para separar efecto desregulación del efecto selección adversa.